src/docfx/build/page/Transformer.cs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Docs.Build
+{
+    internal static class Transformer
+    {
+        public static Func<IEnumerable<DataTypeAttribute>, object, string, object> Transform(List<Error> errors, MarkdownPipelineCallback callback, Document file, JObject extensionData = null)
```

Comment:
>Func<IEnumerable<DataTypeAttribute>, object, string, object> [](start = 22, length = 60)

why return a function?

Reply:
https://github.com/dotnet/docfx/blob/cc3e680a57e42f7f267445883fbdf51340c20fe6/src/docfx/lib/JsonUtility.cs#L184
We need it as input